testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java

```diff
@@ -1367,6 +1367,28 @@ public void testAcidUpdateMajorCompaction()
         });
     }
 
+    @Test(groups = HIVE_TRANSACTIONAL, timeOut = TEST_TIMEOUT)
+    public void testAcidUpdateWithSubqueryAssignment()
+    {
+        // TODO support UPDATE with correlated subquery in assignment
+        withTemporaryTable("test_update_subquery", true, false, NONE, tableName -> {
+            onTrino().executeQuery(format("CREATE TABLE %s (column1 INT, column2 varchar) WITH (transactional = true)", tableName));
+            onTrino().executeQuery(format("INSERT INTO %s VALUES (1, 'x')", tableName));
+            onTrino().executeQuery(format("INSERT INTO %s VALUES (2, 'y')", tableName));
+            onTrino().executeQuery(format("UPDATE %s SET column2 = (SELECT max(name) FROM tpch.tiny.region)", tableName));
+            verifySelectForPrestoAndHive("SELECT * FROM " + tableName, "true", row(1, "MIDDLE EAST"), row(2, "MIDDLE EAST"));
+
+            withTemporaryTable("second_table", true, false, NONE, secondTable -> {
```

Comment:
I am not sure I understand motivation for this sub-testcase. Why source table location would change anything.
Can you add a comment?

Reply:
I remember (or think i remember) there was some problem when working with multiple transactional tables in a query, that's why i added a test case for this.

Will add some comment

Reply:
i had `// UPDATE while reading from another transactional table.` will expand to `// UPDATE while reading from another transactional table. Multiple transactional could interfere with ConnectorMetadata.beginQuery`